crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -537,11 +537,26 @@ where
         .chain_constants(best_block_hash)?
         .confirmation_depth_k();
 
-    let maybe_last_archived_block = find_last_archived_block(
-        client,
-        segment_headers_store,
-        best_block_number.saturating_sub(confirmation_depth_k.into()),
-    )?;
+    // Trying to get the "best block to archive" in both cases: regular and fast sync.
+    let mut best_block_to_archive = best_block_number + 1u32.into();
```

Comment:
Logically this is nonsense, it is not possible to archive block higher than the best block number that exists. And you had to add unnecessary `+1` below just to compensate for this issue that wouldn't exist otherwise.

Reply:
Didn't work how and why?

Reply:
But why? Override is a bad and redundant API. How exactly did it not work without `+1`?